sim/network.go

```diff
@@ -61,24 +61,68 @@ func (n *Network) AddParticipant(p gpbft.Receiver) {
 	n.participants[p.ID()] = p
 }
 
+////// Network interface
+
+func (n *Network) NetworkFor(signer gpbft.Signer, id gpbft.ActorID) *networkFor {
+	return &networkFor{
+		ParticipantID: id,
+		Signer:        signer,
+		Network:       n,
+	}
+}
+
+type networkFor struct {
+	ParticipantID gpbft.ActorID
+	Signer        gpbft.Signer
+	*Network
+}
+
+// Log implements tagging Tracer interface
+func (nf *networkFor) Log(format string, args ...any) {
+	nf.Network.log(TraceLogic, "P%d "+format, append([]any{nf.ParticipantID}, args...)...)
```

Comment:
Thanks. Please enable tracing in some test and check that the logs look reasonable. You've added a prefix here but I didn't spot where the existing prefix has been removed from call sites, so it might be duplicated now.

Reply:
Good catch, thanks. This was in preparation for the ID removal, but after working in #267 I realized with my proposal is no longer needed, so I will remove it from here.

Reply:
Prefix is still here...

Reply:
Removed it from this review, as it adds nothing and it will make more sense once we address the review of my followup PRs. Sorry for the unnecessary noise.